File: Composite/c#/Composite.cs
abstract class Item
{
    protected string ItemName;
    
    protected string OwnerName;
    
    public void SetOwner(string o) => OwnerName = o;
    
    public Item(string name) => ItemName = name;
    
    public virtual void Add(Item subItem) {}
    
    public virtual void Remove(Item subItem) {}
    
    public abstract void Display();
}

class ClickableItem : Item
{
    public ClickableItem(string name) : base(name) {}

    public override void Add(Item subItem)
    {
        throw new Exception();
    }

    public override void Remove(Item subItem)
    {
        throw new Exception();
    }

    public override void Display()
    {
        Console.WriteLine(ItemName);
    }
}

class DropDownItem : Item
{
    private List<Item> _children;

    public DropDownItem(string name) : base(name)
    {
        _children = new List<Item>();
    }

    public override void Add(Item subItem)
    {
        subItem.SetOwner(this.ItemName);
        _children.Add(subItem);
    }

    public override void Remove(Item subItem) => _children.Remove(subItem);

    public override void Display()
    {
        foreach (Item item in _children)
        {
            if (OwnerName != null)
                Console.Write(OwnerName + ItemName);
            item.Display();
        }
    }
}

class Program
{
    static void Main()
    {
        Item file = new DropDownItem("file->");

        Item create = new DropDownItem("create->");
        Item open = new DropDownItem("open->");
        Item exit = new ClickableItem("exit");
        
        file.Add(create);
        file.Add(open);
        file.Add(exit);

        Item project = new ClickableItem("project...");
        Item repository = new ClickableItem("repository...");
        
        create.Add(project);
        create.Add(repository);

        Item solution = new ClickableItem("solution...");
        Item folder = new ClickableItem("folder...");
        
        open.Add(solution);
        open.Add(folder);
        
        file.Display();
        Console.WriteLine();
        
        file.Remove(create);
        
        file.Display();
    }
}

File: Composite/java/Composite.java
import java.util.ArrayList;
import java.util.List;

abstract class Item{
    protected String itemName; 
    protected String ownerName;
    
    public Item(String name){
        itemName = name;
    }
    
    public void setOwner(String o){
        ownerName = o;
    }
    
    public abstract void add(Item subItem) throws Exception;
    
    public abstract void remove(Item subItem) throws Exception;
    
    public abstract void display();
}

class ClickableItem extends Item{
    public ClickableItem(String name){
        super(name);
    }
    
    @Override
    public void add(Item subItem) throws Exception{
        throw new Exception();
    }
    
    @Override
    public void remove(Item subItem) throws Exception{
        throw new Exception();
    }
    
    @Override
    public void display(){
        System.out.println(itemName);
    }
}

class DropDownItem extends Item{
    private List<Item> children;
    
    public DropDownItem(String name){
        super(name);
        children = new ArrayList<>();
    }
    
    @Override
    public void add(Item subItem) throws Exception{
        subItem.setOwner(this.itemName);
        children.add(subItem);
    }
    
    @Override
    public void remove(Item subItem) throws Exception{
        children.remove(subItem);
    }
    
    @Override
    public void display(){
        for (Item child : children){
            if (ownerName != null)
                System.out.print(ownerName + itemName);
            child.display();
        }
    }
}

public class Main {
    public static void main(String[] args) throws Exception{
        Item file = new DropDownItem("file->");

        Item create = new DropDownItem("create->");
        Item open = new DropDownItem("open->");
        Item exit = new ClickableItem("exit");

        file.add(create);
        file.add(open);
        file.add(exit);

        Item project = new ClickableItem("project...");
        Item repository = new ClickableItem("repository...");

        create.add(project);
        create.add(repository);

        Item solution = new ClickableItem("solution...");
        Item folder = new ClickableItem("folder...");

        open.add(solution);
        open.add(folder);

        file.display();
        System.out.println();

        file.remove(create);

        file.display();
    }
}